libraries/node-core-library/src/Async.ts

```diff
@@ -359,6 +359,26 @@ export class Async {
   public static getSignal(): [Promise<void>, () => void, (err: Error) => void] {
     return getSignal();
   }
+
+  /**
+   * Runs a promise with a timeout. If the promise does not resolve within the specified timeout,
+   * it will reject with an error.
+   *
+   * @param promise - The promise to run with a timeout.
+   * @param timeoutMs - The timeout in milliseconds.
+   * @param timeoutMessage - The message to use for the error if the timeout is reached.
+   * @returns A promise that resolves with the result of the input promise, or rejects with a timeout error.
+   */
+  public static runWithTimeoutAsync<TResult>(
+    promise: Promise<TResult>,
+    timeoutMs: number,
+    timeoutMessage: string = 'Operation timed out'
```

Comment:
Take an options object with `action`, `timeoutMs` and `timeoutMessage` params, to match the formatting of `runWithRetries`